vars/findOldestStackVersion.groovy

```diff
@@ -0,0 +1,86 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+/**
+Find the oldest stack version given the condition to compare with.
+
+def version = findOldestStackVersion(versionCondition: "^7.14.0")
+*/
+
+def call(Map args = [:]) {
+  def versionCondition = args.containsKey('versionCondition') ? args.versionCondition : error('findOldestStackVersion: versionCondition parameter is required')
+  def version = versionCondition.substring(1)
```

Comment:
Current implementation only supports the `^` operator for versions. This is the only one used in integrations so far, maybe it should be mentioned in the docs here in case this function is used in other projects.

Reply:
And maybe it would make sense to add something so `if versionCondition[0] != '^'`, it explicitly raises an exception or something saying that the condition is not supported?

Reply:
See https://github.com/elastic/apm-pipeline-library/pull/1248/commits/001a757ad32ba33e035b80f812aeaad5574057fe